app/javascript/widget/mixins/nextAvailabilityTime.js

```diff
@@ -0,0 +1,251 @@
+        this.timeSlot.day !== this.currentDay()
+          ? ''
+          : `at ${this.timeSlot.from}`
+      }`;
+    },
+    hoursAndMinutesLeft() {
+      const { hoursLeft, minutesLeft } = this.hoursAndMinutesBackInOnline;
+
+      const timeLeftChars = [];
+      const languageCode = window.chatwootWebChannel.locale;
+
+      if (hoursLeft > 0) {
+        const roundedUpHoursLeft = minutesLeft > 0 ? hoursLeft + 1 : hoursLeft;
+        const hourRelative = this.generateRelativeTime(
+          roundedUpHoursLeft,
+          'hour',
+          languageCode
+        );
+        timeLeftChars.push(`${hourRelative}`);
+      }
+
+      if (minutesLeft > 0 && hoursLeft === 0) {
+        const roundedUpMinLeft =
+          Math.ceil(minutesLeft / MINUTE_ROUNDING_FACTOR) *
+          MINUTE_ROUNDING_FACTOR;
+        const minRelative = this.generateRelativeTime(
+          roundedUpMinLeft,
+          'minutes',
+          languageCode
+        );
+        timeLeftChars.push(`${minRelative}`);
+      }
+
+      return timeLeftChars.join(' ');
+    },
+    hoursAndMinutesToBack() {
+      const { hoursLeft, minutesLeft } = this.hoursAndMinutesBackInOnline;
+      if (hoursLeft > 3) {
+        return this.exactTimeInAmPm;
+      }
+      if (hoursLeft > 0 || minutesLeft > 0) {
+        return this.hoursAndMinutesLeft;
+      }
+      return 'in some time';
+    },
+    timeLeftToBackInOnline() {
+      if (
+        this.dayDiff > 1 &&
+        this.presentHour() >= this.currentDayTimings.closeHour
+      ) {
+        return `on ${this.dayNameOfNextWorkingDay}`;
+      }
+      if (this.hoursAndMinutesBackInOnline.hoursLeft >= 24) {
+        return `tomorrow`;
+      }
+      return this.hoursAndMinutesToBack;
+    },
+  },
+  mounted() {
+    this.setTimeSlot();
+  },
+  methods: {
+    newDateWithTimeZone() {
+      const date = new Date();
+      const timeZone = this.timeZoneValue;
+      const zonedDate = utcToZonedTime(date, timeZone);
+      return zonedDate;
+    },
+    presentHour() {
+      return this.newDateWithTimeZone().getHours();
+    },
+    presentMinute() {
+      return this.newDateWithTimeZone().getMinutes();
+    },
+    currentDay() {
+      const date = this.newDateWithTimeZone();
+      const day = date.getDay();
+      const currentDay = Object.keys(this.dayNames).find(
+        key => this.dayNames[key] === this.dayNames[day]
+      );
+      return Number(currentDay);
+    },
+    getNextDay(day) {
+      // This code calculates the next day of the week based on the current day. If the current day is Saturday (6), then the next day will be Sunday (0).
+      return (day + 1) % 7;
+    },
+    getNextWorkingHour(day) {
+      const workingHour = this.workingHours.find(
+        slot => slot.day_of_week === day
+      );
+      if (workingHour && !workingHour.closed_all_day) {
+        return workingHour;
+      }
+      return null;
+    },
+    generateRelativeTime(value, unit, languageCode) {
```

Comment:
This can be moved out as a utility function elsewhere, same for `getnextDay`, `newDateWithTimezone`, basically any method that does not access the `data` properties from this mixin directly

Reply:
Let's just move the `generateRelativeTime` method outside, in `shared/helpers/DateHelper.js`